test/metric_dimension/aggregation_dimensions_test.go

```diff
@@ -0,0 +1,147 @@
+	return []string{"mem_used_percent", "cpu_usage_user"}
+}
+
+// getExpectedDimensions returns a list of expected dimensions for the given metric name.
+// This is based on the JSON configuration.
+func getExpectedDimensions(metricName string) [][]string {
+	switch metricName {
+	case "mem_used_percent":
+		return [][]string{
+			{"foo", "bar", "baz", "InstanceId", "InstanceType"},
+			{},
+			{"InstanceId"},
+			{"InstanceId", "InstanceType"},
+			{"foo", "bar", "InstanceType"},
+		}
+	case "cpu_usage_user":
+		return [][]string{
+			{"InstanceId", "InstanceType"},
+			{},
+			{"InstanceId"},
+		}
+	}
+	return nil
+}
+
+func getUnexpectedDimensions(metricName string) [][]string {
+	switch metricName {
+	case "mem_used_percent":
+		return [][]string{
+			{"foo"},
+			{"foo", "bar"},
+			{"InstanceType"},
+		}
+	case "cpu_usage_user":
+		return [][]string{
+			{"foo"},
+			{"foo", "bar"},
+			{"InstanceType"},
+			{"foo", "bar", "baz", "InstanceId", "InstanceType"},
+		}
+	}
+	return nil
+}
+
+func getDimensionValue(dimensionKey string) dimension.ExpectedDimensionValue {
+	switch dimensionKey {
+	case "foo":
+		return dimension.ExpectedDimensionValue{aws.String("fooval")}
+	case "bar":
+		return dimension.ExpectedDimensionValue{aws.String("barval")}
+	case "baz":
+		return dimension.ExpectedDimensionValue{aws.String("bazval")}
+	default:
+		return dimension.UnknownDimensionValue()
+	}
+}
+
+// validate checks that a metric exists with each of the expected dimension sets.
+func (t *AggregationDimensionsTestRunner) validate(metricName string) status.TestResult {
+	r := status.TestResult{
+		Name:   metricName,
+		Status: status.FAILED,
+	}
+	f := metric.MetricValueFetcher{}
+	// Validate the metric name with some expected dimension sets.
+	aggregations := getExpectedDimensions(metricName)
+	for _, aggregation := range aggregations {
+		instructions := []dimension.Instruction{}
+		for _, k := range aggregation {
+			v := getDimensionValue(k)
+			i := dimension.Instruction{Key: k, Value: v}
+			instructions = append(instructions, i)
+		}
+		dd, _ := t.DimensionFactory.GetDimensions(instructions)
+		values, err := f.Fetch("TestAggregationDimensions", metricName, dd,
+			metric.AVERAGE, test_runner.HighResolutionStatPeriod)
+		// Expect values for the metric with the current dimension list.
+		if err != nil || !isAllValuesGreaterThanOrEqualToZero(metricName, values) {
+			return r
+		}
+	}
+
+	// Validate the metric with some dimensions DO NOT EXIST.
+	aggregations = getUnexpectedDimensions(metricName)
+	for _, aggregation := range aggregations {
+		instructions := []dimension.Instruction{}
+		for _, k := range aggregation {
+			v := getDimensionValue(k)
+			i := dimension.Instruction{Key: k, Value: v}
+			instructions = append(instructions, i)
+		}
+		dd, _ := t.DimensionFactory.GetDimensions(instructions)
+		values, _ := f.Fetch("TestAggregationDimensions", metricName, dd,
+			metric.AVERAGE, test_runner.HighResolutionStatPeriod)
```

Comment:
similar to what @khanhntd mentioned in the other place, since this is almost identical to those in the for loop of expectedDimension phase, maybe we can pull this in a separate function to take `aggregation` as input and return `values` and `err`